src/odinapi/views/newdonalettyERANC.py

```diff
@@ -2,297 +2,52 @@
-                    25.09,
-                    24.75,
-                    24.42,
-                    24.10,
-                ]
-                mbarz = np.arange(85, 151, 5)
-                m = np.interp(z, mbarz, mbars)
-                return m
-
-            def g(z, lat):
-                # reference: The Earth's Atmosphere: Its Physics and Dynamics
-                # (Eq. 1.2.4)
-                # e.g 9.80616 [ms^-2] acceleration (gravity) at mean sea level
-                # at 45 degree
-                return (
-                    9.80616
-                    * (
-                        1
-                        - 0.0026373 * np.cos(2 * np.radians(lat))
-                        + 0.0000059 * np.cos(2 * np.radians(lat)) ** 2
-                    )
-                    * (1 - 2.0 * z / geoid_radius(lat))
-                )
-
-            def func(_, z, bspl_eval):
-                return bspl_eval(z)
-
-            def spline(xk, yk, xnew):
-                t_i, c_i, k_i = splrep(xk, yk, k=3)
-                bspl_eval = BSpline(t_i, c_i, k_i)
-                return bspl_eval(xnew)
-
-            newT = spline(z, T, newz)
-            mbar_over_m0 = intermbar(newz) / m0
-            t_i, c_i, k_i = splrep(newz, g(newz, lat) / newT * mbar_over_m0, k=3)
-            bspl_eval = BSpline(t_i, c_i, k_i)
-
-            integral = odeint(func, 0, newz, args=(bspl_eval,))
-            integral = 3.483 * np.squeeze(integral.transpose())
-            integral = newT[1] / newT * mbar_over_m0 * np.exp(-integral)
-
-            normfactor = normrho / spline(newz, integral, normz)
-            rho = normfactor * integral
-            nodens = rho / intermbar(newz) * AVOGADRO / 1e3
-            n2 = wn2 * mbar_over_m0 * nodens
-            o2 = nodens * (mbar_over_m0 * (1 + wo2) - 1)
-            o = 2 * (1 - mbar_over_m0) * nodens
-            o[o < 0] = 0
-            p = nodens * 1e6 * k * newT
-            return newT, p, rho, nodens, n2, o2, o
-
-        # fix Msis from 70-150 km with solar effects
-        msis = M90.Msis90(solardatafile=self.solardatafile)
-        # to ensure an ok stratopause Donal wants to add a 50 km point
-        # from msis
-        msisz = np.r_[49.0, 50.0, 51, np.arange(75, 151, 1)]
-        msisT = msis.extractPTZprofilevarsolar(scan_datetime, lat, lon, msisz)[1]
-        z = np.r_[g5zpt[g5zpt[:, 0] < 60, 0], msisz]
-        temp = np.r_[g5zpt[g5zpt[:, 0] < 60, 2], msisT]
-        normrho = (
-            np.interp([20], g5zpt[:, 0], g5zpt[:, 1])
-            * 28.9644
-            / 1000
-            / Ro
-            / np.interp([20], g5zpt[:, 0], g5zpt[:, 2])
-        )
-        newT, newp, _, _, _, _, _ = intatm(z, temp, newz, 20, normrho[0], lat)
-        zpt = np.vstack((newz, newp, newT)).transpose()
+class ZPT(TypedDict):
+    ScanID: int
+    Z: list[float]
+    P: list[float]
+    T: list[float]
+    latitude: float
+    longitude: float
+    mjd: float
+
+
+def load_zptfile(filepath: str, scanid: int):
```

Comment:
Note to self: Within two weeks this will probably need to be rewritten to work with zarr.